packages/tpetra/core/src/Tpetra_Details_allReduceView.hpp

```diff
@@ -54,13 +54,17 @@
 
 namespace { // (anonymous)
 
-// We can't assume C++14 yet, so we can't have templated constants.
+// helper for detecting views that have Cuda memory
+// Technically, UVM should not be here, but it seems we
+// treat UVM as CudaSpace, so I have left it. With Cuda >= 8 on Linux
+// UVM is handled by page faults in the Kernel, so it should always
+// be addressable.
 template<class ViewType>
-struct view_is_cuda_uvm {
+struct view_uses_cuda_spaces {
```

Comment:
We keep rewriting this "is a CUDA thing" class.  I filed a Kokkos [feature request](https://github.com/kokkos/kokkos/issues/2196).